lib/gollum/app.rb

```diff
@@ -462,7 +462,12 @@ class App < Sinatra::Base
       post '/compare/*' do
         @file     = clean_url(encodeURIComponent(params[:splat].first))
         @versions = params[:versions] || []
-        if @versions.size < 2
+        if @versions.size == 1
+          wikip  = wiki_page(params[:splat].first)
+          commit = wikip.wiki.repo.rev_parse(@versions.first)
```

Comment:
Just for my understanding: the idea is to show the difference between `@version.first` and the latest commit?

Reply:
No. The difference between `@version.first` and its parent.

Reply:
Ah, right! What about the edge case where there is no parent?

Reply:
Good catch. I didn't consider that case at all because it cannot happen in my repo with an empty commit at the beginning. As far as I know, this code fails if the page is created at the very first commit. Even in this case, users can get the same information by clicking the commit ID on the right in the Page History page. So, leaving the error will be an easy option, but redirecting to the commit page will be more user-friendly.